spec/openai/client/files_spec.rb

```diff
@@ -132,24 +132,32 @@
     describe "#fetch_image" do
       let(:cassette) { "files fetch_image" }
       let(:upload_cassette) { "#{cassette} upload" }
+      let(:retrieve_cassette) { "#{cassette} retrieve" }
       let(:filename) { "image.png" }
       let(:file) { File.join(RSPEC_ROOT, "fixtures/files", filename) }
       let(:upload_purpose) { "vision" }
+      let(:response) { OpenAI::Client.new.files.content(id: upload_id) }
+
+      before do
+        # We need to check the file has been processed by OpenAI
+        # before we can delete it.
+        retrieved = VCR.use_cassette(retrieve_cassette) do
+          OpenAI::Client.new.files.retrieve(id: upload_id)
+        end
+        tries = 0
+        until retrieved["status"] == "processed"
+          raise "File not processed after 10 tries" if tries > 10
 
-      def poll_until_processed(max_attempts: 10)
-        VCR.use_cassette("#{cassette}_poll", record: :new_episodes) do
-          max_attempts.times do |attempt|
-            retrieved = OpenAI::Client.new.files.retrieve(id: upload_id)
-            return retrieved if retrieved["status"] == "processed"
-            raise "File not processed after #{max_attempts} attempts" if attempt == max_attempts - 1
+          sleep(1)
```

Comment:
Never liked the `sleep` statements, but it seems like it serves it's purpose, and this is still in the tests so can be refactored later (for example with https://github.com/ruby/timeout)

Reply:
agree. plus behind VCR so it doesn't get hit except when re-recording cassettes

Reply:
Thank you for review!